src/main/java/org/jenkinsci/maven/plugins/hpi/HplMojo.java

```diff
@@ -137,6 +136,9 @@ private void buildLibraries(List<String> paths) throws IOException {
         for (MavenArtifact artifact : artifacts) {
             if(jenkinsPlugins.contains(artifact.getId()))
                 continue;   // plugin dependencies
+            if (artifact.getDependencyTrail().size() < 2) {
+                throw new IllegalStateException("invalid dependency trail");
+            }
             if(artifact.getDependencyTrail().size() >= 1 && jenkinsPlugins.contains(artifact.getDependencyTrail().get(1)))
```

Comment:
It's been wrong for 8 years...

```suggestion
            if(artifact.getDependencyTrail().size() > 1 && jenkinsPlugins.contains(artifact.getDependencyTrail().get(1)))
```

Reply:
I am not touching this pre-existing problem. Instead I added an assertion a line above to make sure the state is as we expect. This assertion passes both before and after this PR.